packages/client/src/client4.ts

```diff
@@ -1928,11 +1928,30 @@ export default class Client4 {
         );
     };
 
-    getPostThread = (postId: string, fetchThreads = true, collapsedThreads = false, collapsedThreadsExtended = false) => {
-        return this.doFetch<PostList>(
-            `${this.getPostRoute(postId)}/thread${buildQueryString({skipFetchThreads: !fetchThreads, collapsedThreads, collapsedThreadsExtended})}`,
+    getPostThread = async (postId: string, fetchThreads = true, collapsedThreads = false, collapsedThreadsExtended = false, direction: 'up'|'down' = 'down', perPage: number = PER_PAGE_DEFAULT, fromCreateAt?: number, fromPost?: string, prevList?: PostList): Promise<PostList> => {
+        const list: PostList = prevList || {
+            order: [postId],
+            posts: {},
+            prev_post_id: '',
+            next_post_id: '',
+        };
+
+        const result = await this.doFetch<PaginatedPostList>(
+            `${this.getPostRoute(postId)}/thread${buildQueryString({skipFetchThreads: !fetchThreads, collapsedThreads, collapsedThreadsExtended, direction, perPage, fromPost, fromCreateAt})}`,
             {method: 'get'},
         );
+
+        list.order!.push(...result.order.slice(1));
+        list.posts = Object.assign(list.posts, result.posts);
+
+        if (result.has_next) {
+            const [nextPostId] = list.order!.slice(-1);
+            const nextPostPointer = list.posts[nextPostId];
+
+            return this.getPostThread(postId, fetchThreads, collapsedThreads, collapsedThreadsExtended, direction, perPage, nextPostPointer?.create_at, nextPostId, list);
```

Comment:
Could you move this logic to the redux action instead of it being part of the Client4 method? We want to keep the client as a simple wrapper around the REST API with more complicated web app-specific logic being in redux

Reply:
sure